src/app/themes.tsx

```diff
@@ -30,19 +70,31 @@ const StyleManager: React.FC<Props> = ({ children }) => {
         return <>{styles}</>;
     });
 
+    const [chosenTheme, setChosenTheme] = useState(theme);
+
     if (typeof window !== "undefined") {
-        return <>{children}</>;
+        return (
+            <ThemeUpdateContext.Provider value={setChosenTheme}>
+                <ThemeProvider theme={chosenTheme}>
+                    <GlobalStyle />
+                    {children}
+                </ThemeProvider>
+            </ThemeUpdateContext.Provider>
+        );
     }
 
     return (
-        <StyleSheetManager sheet={serverStyleSheet.instance} shouldForwardProp={isPropValid}>
-            <ThemeProvider theme={currentTheme}>
-                <>
-                    <GlobalStyle />
+        <ThemeUpdateContext.Provider value={setChosenTheme}>
+            <ThemeProvider theme={chosenTheme}>
+                <GlobalStyle />
+                <StyleSheetManager
```

Comment:
StyleSheetManager has changed position - should it be above ThemeProvider?
(no idea which way around it should be, just you've changed it - can you show that your way is the correct way?)

Reply:
Theme switching only seemed to work in this configuration, not sure why though.